src/components/Courses/CourseCard/tests/CourseCard.test.tsx

```diff
@@ -0,0 +1,113 @@
+import { render } from '@testing-library/react';
+
+import { Provider } from 'react-redux';
+import configureMockStore from 'redux-mock-store';
+
+import { MemoryRouter, Route, Routes } from 'react-router-dom';
+import { CourseCard } from '../CourseCard';
+
+const mockStore = configureMockStore();
+
+jest.mock('../../../../store/server', () => {
+	return {
+		Server: jest.fn().mockImplementation(() => {
+			return {
+				fetchCoursesFromAPI: jest.fn(),
+				getRole: jest.fn(),
+			};
+		}),
+	};
+});
+
+describe('CourseCard', () => {
+	beforeAll(() => {
+		localStorage.setItem('AUTH_TOKEN_REACT_COURSE', 'Bearer test-token');
+	});
+	afterAll(() => {
+		localStorage.removeItem('AUTH_TOKEN_REACT_COURSE');
+	});
+	test('should display title', () => {
+		const { getByText } = renderComponent();
+
+		const title = getByText('React course');
+
+		expect(title).toBeInTheDocument();
+	});
+
+	test('should display description', () => {
+		const { getByText } = renderComponent();
+		const description = getByText('description for React course');
+
+		expect(description).toBeInTheDocument();
+	});
+
+	test('should display duration in the correct format', () => {
+		const { getByText } = renderComponent();
+		const duration = getByText('02:03 hours');
+
+		expect(duration).toBeInTheDocument();
+	});
+
+	test('should display authors list', () => {
+		const { getByText } = renderComponent();
+		const authors = getByText('Rita, John');
+
+		expect(authors).toBeInTheDocument();
+	});
+
+	test('should display created date in the correct format', () => {
+		const { getByText } = renderComponent();
+		const creationDate = getByText('9.3.2021');
+
+		expect(creationDate).toBeInTheDocument();
+	});
+
+	function renderComponent() {
+		const initUserState = {

```

Comment:
It is a good idea to move state object to a separate mock file and import it where needed